pkg/placement/controllers/scheduling/scheduling_controller.go

```diff
@@ -517,7 +517,9 @@ func (c *schedulingController) generateDecisionGroups(
 
 	// The rest of the clusters will also be put into decision groups.
 	var matched []clusterapiv1beta1.ClusterDecision
-	for _, cluster := range clusterNames.UnsortedList() {
+	clusterNameList := clusterNameSet.UnsortedList()
+	sort.Strings(clusterNameList)
+	for _, cluster := range clusterNameList {
 		matched = append(matched, clusterapiv1beta1.ClusterDecision{
```

Comment:
Not sure if sorting the `matched` is better, because if the `matched` get some results from `matched, status := filterClustersBySelector(d.ClusterSelector, clusters, clusterNameSet)`, are they in order？

Reply:
Good suggestion. Need to sort all the matched before dividing into groups.